packages/rancher-monitoring/generated-changes/patch/values.yaml.patch

```diff
@@ -526,15 +526,25 @@
    ## Deploy default dashboards.
    ##
    defaultDashboardsEnabled: true
-@@ -530,6 +936,7 @@
+ 
++  # Additional options for defaultDashboards
++  defaultDashboards:
++    namespace: cattle-dashboards
```

Comment:
This is somewhat of a breaking change.

A user who overrode `grafana.sidecar.searchNamespace` before to something non-default (e.g. `my-namespace`) would see the current Helm chart ignore that override on an upgrade (since they would need to modify `grafana.defaultDashboards.namespace` instead now).

However, I think it's highly unlikely anyone modified the name of the default namespace in which dashboards are deployed, so I don't think this is a risky move.

Reply:
Are there steps that could be documented that a user could follow to recover their dashboards in the event they did the very unlikely thing?

Reply:
If a user was just using the default dashboards, then there's nothing to recover since it will be recreated. If they modified default dashboards in place, then that's not supported since they are trying to manage a Helm managed resource outside of Helm.

However, I'm not sure what would happen if they created non-Helm managed resources in a Helm managed namespace resource that then got deleted. I'll try to look into what happens in that case and get back to you.

Reply:
@brendarearden just tried it out, I think that I need to refactor this PR to ensure the cattle-dashboards namespace never gets deleted unless you entirely uninstall the chart. We should also probably add something in the docs since we currently tell users to place dashboards in cattle-dashboards; as a result, those would get wiped out on a Helm uninstall / reinstall.

Reply:
@brendarearden I added the docs to https://github.com/rancher/docs/issues/3125 w.r.t. what you can do to prevent dashboards from being deleted in the current monitoring chart (workaround is adding a Helm resource annotation to the namespace manually) and added a flag to `cleanupOnUninstall` that defaults to false. 

When set, a helm resource policy is set to keep the namespace, which prevents the namespace from being destroyed on a helm uninstall.